polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -614,8 +706,10 @@ impl<T: Config> Pallet<T> {
 			<T as Config>::Currency::unreserve(&who, rebate);
 		};
 		let info = ParaInfo { manager: who.clone(), deposit, locked: None };
+		let stash_info = StashInfo { stash: who.clone(), pending_refund: None };
```

Comment:
It seems sensible to merge this with `ParaInfo`? And I would suggest making the stash account an `Option` - it has to be set explicitly. We really need to have some form of "contract" ensuring that it is ok to charge this account. An extrinsic, with a descriptive name, that makes it crystal clear that this account will get charged.

Reply:
The upgrade will then fail, if such an account has not been set before. I wonder whether just returning an error suffices here ... maybe we should trigger an event for better visibility of what went wrong? :thinking: